core/objects.go

```diff
@@ -347,6 +348,13 @@ func (ch Challenge) checkConsistency() error {
 	return nil
 }
 
+func (ch Challenge) GenerateID() string {
```

Comment:
In a previous PR I think we about calling this `StringID()` - want to do that now? And below would be `FindChallengeByStringID`